lib/handle-pull-request-change.js

```diff
@@ -3,29 +3,38 @@ module.exports = handlePullRequestChange
 const isSemanticMessage = require('./is-semantic-message')
 const getConfig = require('probot-config')
 
-async function commitsAreSemantic (context) {
+async function commitsAreSemantic (context, allCommits = false) {
   const commits = await context.github.pullRequests.getCommits(context.repo({
     number: context.payload.pull_request.number
   }))
 
   return commits.data
-    .map(element => element.commit)
-    .some(commit => isSemanticMessage(commit.message))
+    .map(element => element.commit)[allCommits ? 'every' : 'some'](commit => isSemanticMessage(commit.message))
 }
 
 async function handlePullRequestChange (context, config) {
   const { title, head } = context.payload.pull_request
-  const { titleOnly } = getConfig(context, 'semantic.yml')
+  const { titleOnly, commitsOnly } = await getConfig(context, 'semantic.yml')
   const hasSemanticTitle = isSemanticMessage(title)
-  const hasSemanticCommits = await commitsAreSemantic(context)
-  const isSemantic = titleOnly
-    ? hasSemanticTitle
-    : hasSemanticTitle || hasSemanticCommits
+  const hasSemanticCommits = await commitsAreSemantic(context, commitsOnly)
+
+  let isSemantic
+
+  if (titleOnly) {
+    isSemantic = hasSemanticTitle
+  } else if (commitsOnly) {
+    isSemantic = hasSemanticCommits
+  } else {
+    isSemantic = hasSemanticTitle || hasSemanticCommits
+  }
+
   const state = isSemantic ? 'success' : 'pending'
 
   function getDescription () {
     if (hasSemanticTitle) return 'ready to be squashed'
     if (hasSemanticCommits) return 'ready to be merged or rebased'
+    if (titleOnly) return 'add a semantic PR title'
+    if (commitsOnly) return 'add a semantic commit'
```

Comment:
> add a semantic commit

Shouldn't this message be something more like "make sure every commit is semantic"?

This gets at the root of the trouble with this new `commitsOnly` options: People will have to amend their commit history in order to comply with it. Many people don't know how to do that. I guess the challenge here is to convey that in a very short status message...

Reply:
Really good point. I've updated the wording as suggested.